crates/oxc_transformer/src/helpers/module_imports.rs

```diff
@@ -116,9 +118,13 @@ impl<'a> ModuleImports<'a> {
         &self,
         source: Atom<'a>,
         names: std::vec::Vec<NamedImport<'a>>,
+        ctx: &mut TraverseCtx<'a>,
     ) -> Statement<'a> {
         let var_kind = VariableDeclarationKind::Var;
-        let callee = self.ast.expression_identifier_reference(SPAN, Atom::from("require"));
+        let ident =
+            ctx.create_reference_id(SPAN, Atom::from("require"), None, ReferenceFlag::read());
```

Comment:
Strictly speaking, we should also handle if there's a binding in root scope for a var called `require` e.g. `let require = 123;`. This would create output that doesn't work properly, but at least scopes tree will be correct!

```suggestion
        let symbol_id = ctx.scopes().get_root_binding("require");
        let ident =
            ctx.create_reference_id(SPAN, Atom::from("require"), symbol_id, ReferenceFlag::read());
```

Reply:
Good catch!